File: 프로그래머스/Level2/짝지어 제거하기.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int solution(string s){
    int answer = 0;
    stack<char> stk;
   
    for(int i=0; i<s.length(); i++){
        if(stk.size() > 0 && stk.top() == s[i]){
            stk.pop();
        }
        else{
            stk.push(s[i]);
        }
    }
    
    if(stk.empty()) return 1;
    return answer;
}

File: 프로그래머스/Level2/짝지어 제거하기.js
function solution(s)
{
    const stack = [];
    stack.push(s[0]);
    
    for(let i=1; i<s.length; i++){
        if(s[i] === stack[stack.length-1])
            stack.pop();          
        else 
            stack.push(s[i]);
    }
  
    return stack.length === 0 ? 1 : 0;
}
